docs/reference/pages/model_recipes.rst

```diff
@@ -196,3 +196,31 @@ Here, ``blogs.filter(tags__name=tag)`` invokes a reverse Django QuerySet filter
 Iterating through ``page.tags.all`` will display each tag associated with ``page``, while the link(s) back to the index make use of the filter option added to the ``BlogIndexPage`` model. A Django query could also use the ``tagged_items`` related name field to get ``BlogPage`` objects associated with a tag.
 
 This is just one possible way of creating a taxonomy for Wagtail objects. With all of the components for a taxonomy available through Wagtail, you should be able to fulfil even the most exotic taxonomic schemes.
+
+
+Have redirects created automatically when changing page slug
+------------------------------------------------------------
+
+You may want redirects created automatically when a url gets changed in the admin so as to avoid broken links. You can add something like the following block to `/models/wagtail_hooks.py`.
+
+
+.. code-block:: python
+
+    from wagtail.core import hooks
+    from wagtail.contrib.redirects.models import Redirect
+
+    # Create redirect when editing slugs
+    @hooks.register('before_edit_page')
+    def create_redirect_on_slug_change(request, page):
+        try:
+            if page.slug != request.POST['slug']:
+                Redirect.objects.create(
+                        old_path=page.url[:-1],
+                        site=page.get_site(),
+                        redirect_page=page
+                    )
+        except:
```

Comment:
I'd rather not encourage people to use blanket `try` / `except` blocks, as they risk hiding legitimate errors that the developer should know about. Is there a specific exception that this code is likely to throw (and can be legitimately ignored)? If so, we should specify that exception in the `except`.

Reply:
Yes agreed. That was initially there since this hook is called both when opening the Page edit admin page (GET request), and also when saving the page (POST), so this was an easy (bad) way to distinguish between them, as well as suppressing any unseen errors.

I have altered that, I hope that’s more appropriate?

https://github.com/wagtail/wagtail/pull/5002/files#diff-75e7c538654bfd89118b597d9090b334